packages/modules/workflow-engine-inmemory/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -148,7 +156,103 @@ export class InMemoryDistributedTransactionStorage
     }
 
     if (hasFinished) {
-      this.storage.delete(key)
+      const tenMinutes = 10 * 60 * 1000
```

Comment:
As we talked with Carlos on a separate Slack thread, unless the user has marked a workflow idempotent, we should clear it from storage as soon as it finishes.

This can of course be done as part of a separate PR, as I assume the storage removal should also be applied elsewhere.

Reply:
I agree, but here it is more in the scenario where you have concurrent execution due to retryal on awaiting for example where you need to have the state while the concurrent one get to that point. But we can reduce it a bit, it is for the inmemory to prevent any subsequent execution and prevent another type of race condition :)

Reply:
We use the in-memory for the tests so the tests won't work if what I described doesn't work the same way as the Redis engine.

The rollback is fine, that's not an issue at all. Would the `await myWorkflow(container).cancel(transaction)` resolve only once the rollback is done? or it won't wait for the rollback?

Reply:
It works the same way as the redis does. 
for your question about the cancel, we will introduce a cancel method as part of the engine which is missing at the moment, but otherwise it works like for the run, if it is an async workflow (with async compensation) then you will have an acknowledgment otherwise it will have finished the rollback when the promise resolve.
and this cancel will be part of this pr 👍

Reply:
@sradevski we will generate a snapshot for you to test, but we should be good now. As mentioned, we delete the memory storage immediately after finishing an execution to completion (success/reverted). Any other execution would be considered as superfluous and be discarded right away. In regards to the cancellation is everything ok in the understanding of its execution?